### PR TITLE
test: chat screen ui

### PR DESCRIPTION
**Pull request by IskandarGhobril**
This PR enhances the testing coverage and reliability of the `ChatScreen` UI.

**Key Changes:**

1. **UI Test Enhancements for `ChatScreen`:**
   - Added tests to ensure the visibility of the `ChatScreen` top app bar, message list, and navigation actions.
   - Created specific tests to verify navigation actions:
     - Back button navigates back.
     - "Record Response" button navigates to `Screen.SPEAKING`.
     - "Request Feedback" button navigates to `Screen.FEEDBACK`.

2. **Modified `ChatScreen`:**
   - Added test tags to multiple composables to ensure they are correctly displayed in the UI.

This PR ensures the `ChatScreen` UI performs reliably across different states and that it is well implemented with the screen's navigation, loading, and message display functionality.

### CODE REVIEW

File: app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt [hunk 1]
```diff
@@ -19,183 +19,153 @@ import com.github.se.orator.model.chatGPT.ChatViewModel
 import com.github.se.orator.ui.navigation.NavigationActions
 import com.github.se.orator.ui.navigation.Screen
 import com.github.se.orator.ui.network.Message
-import com.github.se.orator.ui.theme.AppColors // Import AppColors
-import com.github.se.orator.ui.theme.AppDimensions // Import AppDimensions
+import com.github.se.orator.ui.theme.AppColors
+import com.github.se.orator.ui.theme.AppDimensions
 
-/**
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt:25]: I don't feel like the doc should have been deleted.

File: app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt [hunk 2]
```diff
@@ -19,183 +19,153 @@ import com.github.se.orator.model.chatGPT.ChatViewModel
...
- * top app bar with a back button, a scrollable list of messages, and buttons at the bottom for
- * recording a response or requesting feedback.
- *
- * @param navigationActions An instance of [NavigationActions] to handle navigation events.
- * @param chatViewModel The [ChatViewModel] that provides chat messages and loading state.
- */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ChatScreen(navigationActions: NavigationActions, chatViewModel: ChatViewModel) {
-  // Collect the list of chat messages from the view model as a state.
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt:38]: Same comment as above, the comments were nice.

File: app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt [hunk 3]
```diff
@@ -19,183 +19,153 @@ import com.github.se.orator.model.chatGPT.ChatViewModel
...
+                              text = "Request Feedback",
+                              modifier = Modifier.testTag("request_feedback_button_text"))
+                        }
+                  }
+            }
+      },
+      modifier = Modifier.testTag("scaffold"))
 }
 
-/**
```
Comment (MathisHage) [inline app/src/main/java/com/github/se/orator/ui/overview/ChatScreen.kt:163]: Docs deleted here as well ?

**Review by MathisHage — CHANGES_REQUESTED** [review]
Everything looks fine, I am just requesting changes for the documentation, which I don't get why some of it got deleted.

**Comment by IskandarGhobril** [comment]
@MathisHage I added back the comments. Do you want to check them out?

**Comment by MathisHage** [comment]
Perfect, good for merging (after the CI finishes :wink: )

**Review by MathisHage — APPROVED** [review]

**Comment by IskandarGhobril** [comment]
I removed some lines from the theme files because they were causing conflicts with the CI. @MathisHage pointed out that it's possible due to the difference between the versions of the gradle which seems quite logical.

**Pull request merged**